src/main/java/de/tum/in/www1/artemis/service/connectors/localci/LocalCIResultProcessingService.java

```diff
@@ -141,6 +146,8 @@ public void processResult() {
                     if (participation.getProgrammingExercise() == null) {
                         participation.setProgrammingExercise(programmingExerciseRepository.getProgrammingExerciseFromParticipationElseThrow(participation));
```

Comment:
Since we already potentially load the exercise from the database here, can you make sure that it contains the configuration, so we don't have to make two database calls?